stub_uploader/get_version.py

```diff
@@ -49,8 +50,8 @@ def ensure_specificity(ver: list[int], specificity: int) -> None:
     ver.extend([0] * (specificity - len(ver)))
 
 
-def compute_incremented_version(
-    version_spec: str, published_versions: list[Version]
+def compute_stub_version(
+    version_spec: str, published_versions: list[Version], date: datetime.date
```

Comment:
ignorable nit: should we try to give a more specific error if given date isn't in the future? (I know we'll error on the `assert new_version > max_published`, but that is a sort of catch-all)

Reply:
We call it once with the current date, and it would make unit testing harder, so I would prefer not to check this here.